src/items.rs

```diff
@@ -408,6 +408,122 @@ impl<'a> FmtVisitor<'a> {
         }
     }
 
+    pub fn visit_enum(&mut self,
+                      ident: ast::Ident,
+                      vis: ast::Visibility,
+                      enum_def: &ast::EnumDef,
+                      generics: &ast::Generics,
+                      span: Span)
+    {
+        // Empty enums are a bit weird (and not like empty structs), let's not touch them for now
+        if enum_def.variants.len() == 0 {
+            let result = self.snippet(span);
+            self.changes.push_str_span(span, &result);
+            return;
+        }
+
+        let header_str = self.format_header("enum", ident, vis);
+        self.changes.push_str_span(span, &header_str);
+
+        let generics_buf = self.format_generics(generics, enum_def.variants[0].span.lo);
+        self.changes.push_str_span(span, &generics_buf);
+
+        let enum_snippet = self.snippet(span);
+        // FIXME this will give incorrect results if there is a { in a commet.
+        self.last_pos = span.lo + BytePos(enum_snippet.find('{').unwrap() as u32 + 1);
+
+        self.block_indent += config!(tab_spaces);
+        for (i, f) in enum_def.variants.iter().enumerate() {
+            let next_span_start: BytePos = if i == enum_def.variants.len() - 1 {
+                span.hi
+            } else {
+                enum_def.variants[i + 1].span.lo
+            };
+
+            self.visit_variant(f, i == enum_def.variants.len() - 1, next_span_start);
+        }
+        self.block_indent -= config!(tab_spaces);
+
+        self.format_missing_with_indent(span.lo + BytePos(enum_snippet.rfind('}').unwrap() as u32));
+        self.changes.push_str_span(span, "}");
+    }
+
+    // Variant of a enum
+    fn visit_variant(&mut self,
+                     field: &ast::Variant,
+                     last_field: bool,
+                     next_span_start: BytePos)
+    {
+        if self.visit_attrs(&field.node.attrs) {
+            return;
+        }
+
+        if let ast::VariantKind::TupleVariantKind(ref types) = field.node.kind {
+            self.format_missing_with_indent(field.span.lo);
+
+            let vis = match field.node.vis {
+                ast::Visibility::Public => "pub ",
+                ast::Visibility::Inherited => ""
+            };
+
+            let mut header = vis.to_owned();
+            header.push_str(&field.node.name.to_string());
```

Comment:
Probably better to push 3 times to self.changes, rather than pushing to header and then pushing header (I realise I probably did this in other places, but I think it would be better not to, with hindsight)